exercises/matthew_bowles/ex02_oo_basics/inc/Shape.h

```diff
@@ -0,0 +1,26 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+
+#include <string>
+
+
+class Shape
+{
+public:
+	Shape(std::string type, int sideCount);
+	virtual ~Shape();
+
+	std::string getType() const;
+	int getSideCount() const;
+	virtual double calcArea() const;
+	virtual double calcPerimeter() const;
+	virtual void print() const;
```

Comment:
I should have mentioned this the first time round, but this class should be pure abstract class, ie methods such as `calcArea` should not have a dummy implementation (rather they should be `virtual calcArea() const = 0;. You should not allow `Shape` to be created it self.